src/skottie-player.ts

```diff
@@ -141,54 +440,368 @@ export class SkottiePlayer extends LitElement {
+        this._updateLayerVisibility(i);
+        this._updateLayerColor(i);
+      }
+    }
+  }
+
+  /**
+   * Updates the visibilty of the layer with given index in the layer array.
+   */
+  _updateLayerVisibility(layerIndex: number): void {
+    if (this.shadowRoot && this.content.layers) {
+      const layerVisibility = this.shadowRoot.getElementById(this.content.layers[layerIndex].nm + 
+        ' Visibility') as HTMLInputElement;
+      if (layerVisibility) {
+        if (layerVisibility.checked) {
+          this.content.layers[layerIndex].ks.o.k = 100;
+        } else {
+          this.content.layers[layerIndex].ks.o.k = 0;
+        }
+      }
+    }
+  }
+
+  /**
+   * Updates the color of the layer with given index in the layer array.
+   */
+  _updateLayerColor(layerIndex: number): void {
+    if (this.shadowRoot && this.content.layers) {
+      if (this.content.layers[layerIndex].ty === 1) {
+        const colorInput = this.shadowRoot.getElementById(this.content.layers[layerIndex].nm + 
+            ' Color Input') as HTMLInputElement;
+        if (colorInput) {
+          this.content.layers[layerIndex].sc = colorInput.value;
+        }
+      } else if (this.content.layers[layerIndex].ty === 4) {
+        for (let shapeIndex = 0; 
+          shapeIndex < (this.content.layers[layerIndex].shapes as any).length;
+          shapeIndex++) {
+          const colorInput = this.shadowRoot.getElementById(this.content.layers[layerIndex].nm + 
+              ' ' + (this.content.layers[layerIndex].shapes as any)[shapeIndex].nm +
+              ' Color Input') as HTMLInputElement;
+          if (colorInput) {
+            this._setShapeColor((this.content.layers[layerIndex].shapes as any)[shapeIndex],
+              this._hexToRgba(colorInput.value));
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Sets the color of a given shape.
+   * @param shape   Type can be fill or group
+   * @param rgba    Has a length of 4, values in range [0, 1]
+   */
+  _setShapeColor(shape: any, rgba: Array<number>): void {
+    if (shape.ty === 'gr' && shape.it) {
+      for (let i = 0; i < shape.it.length; i++) {
+        if (shape.it[i].ty === 'fl') {
+          shape.it[i].c.k = rgba;
+          return;
+        }
+      }
+    }
+  }
+
+  /**
+   * Converts a hex string to a rgba array.
+   */
+  _hexToRgba(hex: string): Array<number> {
+    const r = parseInt(hex.slice(1, 3), 16) / 255;
+    const g = parseInt(hex.slice(3, 5), 16) / 255;
+    const b = parseInt(hex.slice(5, 7), 16) / 255;
+    return [r, g, b, 1];
+  }
+
+  /**
+   * Gets the number of image assets from an array of required assets.
+   */
+  _getNumAssets(assets: Array<any>): number {
+    let length = 0;
+    for (let i = 0; i < assets.length; i++) {
+      if(assets[i].p != undefined) {
+        length++;
+      }
+    }
+    return length;
+  }
+
+  /**
+   * Determines if the animation should be rendered.
+   */
+  _shouldRenderAnimation(): boolean {
```

Comment:
I believe the google style guide changed to discourage the _ for private methods, instead mark the function as private

Reply:
Update the other functions